pkg/controllers/managementuser/rbac/prtb_handler.go

```diff
@@ -176,13 +270,79 @@ func (p *prtbLifecycle) ensurePRTBDelete(binding *v3.ProjectRoleTemplateBinding)
 		}
 	}
 
+	if err := p.ensurePSAPermissionsDelete(binding); err != nil {
+		return fmt.Errorf("error deleting psa RBAC resources: %w", err)
+	}
+
 	if err := p.m.deleteServiceAccountImpersonator(binding.UserName); err != nil {
 		return fmt.Errorf("error deleting service account impersonator: %w", err)
 	}
 
 	return p.reconcileProjectAccessToGlobalResourcesForDelete(binding)
 }
 
+// ensurePSAPermissionsDelete removes the ClusterRole and ClusterRoleBinding
+// that were created to grant 'updatepsa' permissions when a ProjectRoleTemplateBinding
+// is deleted. This function only performs cleanup if the binding's RoleTemplate
+// actually granted updatepsa permissions, ensuring proper cleanup of PSA-related
+// RBAC resources without affecting unrelated bindings.
+func (p *prtbLifecycle) ensurePSAPermissionsDelete(binding *v3.ProjectRoleTemplateBinding) error {
+	psaRec := authorizer.AttributesRecord{
+		Verb:            "updatepsa",
+		APIGroup:        management.GroupName,
+		Resource:        v32.ProjectResourceName,
+		Name:            binding.ProjectName,
+		ResourceRequest: true,
+	}
+	_, projectName, found := strings.Cut(binding.ProjectName, ":")
+	if !found {
+		return fmt.Errorf("invalid project name format")
+	}
+	prtbRoleTemplate, err := p.rtLister.Get("", binding.RoleTemplateName)
+	if err != nil {
+		return err
+	}
+
+	if rbac.RulesAllow(psaRec, prtbRoleTemplate.Rules...) {
+		// delete the ClusterRoleBinding first
+		// if ClusterRole is not used by other ClusterRoleBindings:
+		// then delete the ClusterRole
+		// skip the deletion otherwhise
+		psaCRBName := fmt.Sprintf("%s-updatepsa", projectName)
+		err = p.crbClient.Delete(psaCRBName, &metav1.DeleteOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("error deleting clusterrolebinding %v: %w", psaCRBName, err)
+		}
+		psaCRName := fmt.Sprintf("%s-namespaces-psa", projectName)
+		isUsed := p.isClusterRoleUsed(psaCRName)
+		if !isUsed {
+			err = p.crClient.Delete(psaCRName, &metav1.DeleteOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				return fmt.Errorf("error deleting clusterrole %v: %w", psaCRName, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// IsClusterRoleUsed checks if a ClusterRole is referenced by any ClusterRoleBinding
+func (p *prtbLifecycle) isClusterRoleUsed(clusterRoleName string) bool {
+	bindings, err := p.crbClient.List(metav1.ListOptions{})
+	if err != nil {
+		return false
+	}
+
+	var usingBindings []string
+	for _, binding := range bindings.Items {
+		if binding.RoleRef.Kind == "ClusterRole" && binding.RoleRef.Name == clusterRoleName {
+			usingBindings = append(usingBindings, binding.Name)
```

Comment:
nit: you can simply return true here as we don't care about how many times it is used